From c1c832f83f8c62ce79cb1345cccd8ec40c4b0c7b Mon Sep 17 00:00:00 2001
From: Thomas Aven <thomas.aven@arm.com>
Date: Fri, 13 Mar 2026 09:11:25 +0100
Subject: [PATCH 2/6] Add missing conditional rendering functionality check

Conditional rendering is used in some VK_KHR_copy_memory_indirect tests,
but did not check for support.

Components: Vulkan

VK-GL-CTS issue: 6398

Affected tests:
dEQP-VK.api.copy_and_blit.*conditional*

Change-Id: I64a29250a86ff8a5d9ee5678aefab9a871f46a40
Upstream-Status: Pending
---
 .../modules/vulkan/api/vktApiCopiesAndBlittingTests.cpp      | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/external/vulkancts/modules/vulkan/api/vktApiCopiesAndBlittingTests.cpp b/external/vulkancts/modules/vulkan/api/vktApiCopiesAndBlittingTests.cpp
index f4e8ee650..dc0d6f2ce 100644
--- a/external/vulkancts/modules/vulkan/api/vktApiCopiesAndBlittingTests.cpp
+++ b/external/vulkancts/modules/vulkan/api/vktApiCopiesAndBlittingTests.cpp
@@ -17320,6 +17320,11 @@ public:
             }
         }
         checkFormatSupport(context, m_params.dst.image.format);
+
+        if (m_params.useConditionalRender)
+        {
+            context.requireDeviceFunctionality("VK_EXT_conditional_rendering");
+        }
     }
 
 private:
-- 
2.43.0

